drivers/digital-io/max22196/max22196.c

```diff
@@ -0,0 +1,557 @@
+ * @param desc - MAX22196 device descriptor.
+ * @param ch - Channel for which the channel counter is read.
+ * @param cnt_msb_lsb_bytes - 8MSBs represent the MSB byte read from the specifc
+ *			      chanel's counter, 8LSBs represent the LSB byte.
+ * @return 0 in case of succes, negative error code otherwise.
+*/
+int max22196_get_chan_cnt(struct max22196_desc *desc, uint32_t ch,
+			  uint16_t *cnt_msb_lsb_bytes)
+{
+	int ret;
+	uint32_t reg_val;
+
+	if (ch > MAX22196_CHANNELS - 1)
+		return -EINVAL;
+
+	ret = max22196_reg_update(desc, MAX22196_START_STOP_REG,
+				  MAX22196_CNT_MASK(ch),
+				  no_os_field_prep(MAX22196_CNT_MASK(ch), 0));
+	if (ret)
+		return ret;
+
+	ret = max22196_reg_read(desc, MAX22196_CNT_LSB_REG(ch), &reg_val);
+	if (ret)
+		return ret;
+
+	*cnt_msb_lsb_bytes = no_os_field_prep(MAX22196_LSB_MASK, reg_val);
+
+	ret = max22196_reg_read(desc, MAX22196_CNT_MSB_REG(ch), &reg_val);
+	if (ret)
+		return ret;
+
+	*cnt_msb_lsb_bytes |= no_os_field_prep(MAX22196_MSB_MASK, reg_val);
+
+	return max22196_reg_update(desc, MAX22196_START_STOP_REG,
+				   MAX22196_CNT_MASK(ch),
+				   no_os_field_prep(MAX22196_CNT_MASK(ch), 1));
+}
+
+/**
+ * @brief Initialize and configure the MAX22196 device.
+ * @param desc - device descriptor for the MAX22196 that will be initialized.
+ * @param param - initialization parameter for the device.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int max22196_init(struct max22196_desc **desc,
+		  struct max22196_init_param *param)
+{
+	struct max22196_desc *descriptor;
+	uint8_t reg_val;
+	int ret;
+
+	descriptor = no_os_calloc(1, sizeof(*descriptor));
+	if (!descriptor)
+		return -ENOMEM;
+	ret = no_os_spi_init(&descriptor->comm_desc, param->comm_param);
+	if (ret)
+		goto error;
+
+	if (param->crc_en) {
+		ret = no_os_gpio_get(&descriptor->crc_desc, param->crc_param);
+		if (ret)
+			goto spi_error;
+
+		ret = no_os_gpio_direction_output(descriptor->crc_desc,
+						  NO_OS_GPIO_HIGH);
+		if (ret)
+			goto gpio_error;
+
+		descriptor->crc_en = true;
+	}
+
+	/* Clear the latched faults generated at power-up of MAX22196. */
+	ret = max22196_reg_read(descriptor, MAX22196_FAULT1_REG, &reg_val);
+	if (ret)
+		goto gpio_error;
+
+	ret = max22196_reg_read(descriptor, MAX22196_FAULT2_REG, &reg_val);
+	if (ret)
+		goto gpio_error;
+
+	*desc = descriptor;
+
+	return 0;
+gpio_error:
+	if (param->crc_en)
+		no_os_gpio_remove(descriptor->crc_desc);
+spi_error:
+	no_os_spi_remove(descriptor->comm_desc);
+error:
+	no_os_free(descriptor);
+
+	return ret;
+}
+
+/**
+ * @brief Free the resources allocated during init.
+ * @param desc - device descriptor for the MAX14916 that will be initialized.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int max22196_remove(struct max22196_desc *desc)
+{
+	int ret, i;
+
+	if (!desc)
+		return -ENODEV;
+
+	for (i = 0; i < MAX22196_CHANNELS; i++) {
+		ret = max22196_set_chan_cnt(desc, i, 0);
```

Comment:
in max22190 you are using a macro instead of 0 value, let's try to keep things as similar as possible between the two driver